middleware/varnish/detectors-varnish.tf

```diff
@@ -0,0 +1,135 @@
+# failed backend in varnish detection
+resource "signalfx_detector" "varnish_backend_failed" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Varnish Backend Failed"
+
+  program_text = <<-EOF
+    signal = data('varnish.backend_fail', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('signal')
+    detect(when(signal > ${var.varnish_backend_failed_threshold_critical})).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is too high > ${var.varnish_backend_failed_threshold_critical}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.detectors_disabled)
+    notifications         = coalescelist(var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+# varnish backend req
+resource "signalfx_detector" "varnish_backend_req" {
```

Comment:
I am not sure this detector is relevant. no matter the amount of traffic on varnish it depends on the platform and it is not an anomaly itself.

comparison with past value (https://docs.signalfx.com/en/latest/detect-alert/alert-condition-reference/hist-anomaly.html) could make it more useful. 

if you really want this detector for specific usage so you at least need to disable it by default while it could not work for generic purpose